### PR TITLE
[CARBONDATA-3893] [IUD] Fix getting block name in compacted segment with dot for horizontal compaction delta files

### PR DESCRIPTION
**Pull request by Zhangshunyu**
### Why is this PR needed?
 Currentlly, during horizontal compaction, driver will list delta files in segment to merge them, during this comparasion it will use block name, current code only consider the blocks in normal segment, but not consider the compacted segement, for example, in XXXX-12.2-time.deletedelta the block name should be XXXX-12.2 not XXXX, if use XXX it will not list the delta files and horizontal compaction will not merge delta files.
 
 ### What changes were proposed in this PR?
Correct the parse of block name 
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3333/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1593/

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java [hunk 1]
```diff
@@ -450,8 +450,7 @@ public boolean accept(CarbonFile pathName) {
             String fileName = pathName.getName();
             if (fileName.endsWith(CarbonCommonConstants.DELETE_DELTA_FILE_EXT)
                 && pathName.getSize() > 0) {
-              String firstPart = fileName.substring(0, fileName.indexOf('.'));
-              String blkName = firstPart.substring(0, firstPart.lastIndexOf("-"));
+              String blkName = fileName.substring(0, fileName.lastIndexOf("-"));
```
Comment (Indhumathi27) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java:453]: Can you please add a testcase to check num of files after update/delete in a compacted segment

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1598/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3338/

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java [hunk 2]
```diff
@@ -450,8 +450,7 @@ public boolean accept(CarbonFile pathName) {
             String fileName = pathName.getName();
             if (fileName.endsWith(CarbonCommonConstants.DELETE_DELTA_FILE_EXT)
                 && pathName.getSize() > 0) {
```
Comment (marchpure) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java:452]: if ( pathName.getSize() > 0 && fileName.endsWith(CarbonCommonConstants.DELETE_DELTA_FILE_EXT))  
will be better

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3342/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1602/

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java (hunk 1 above)
Comment (Zhangshunyu) [reply core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java:453]: @Indhumathi27 added test case

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3343/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1603/

File: core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java [hunk 3]
```diff
@@ -471,6 +471,13 @@ public static void cleanUpDeltaFiles(CarbonTable table, boolean forceDelete) thr
     LoadMetadataDetails[] details =
         SegmentStatusManager.readLoadMetadata(table.getMetadataPath());
 
+    SegmentUpdateStatusManager updateStatusManager = new SegmentUpdateStatusManager(table);
+    SegmentUpdateDetails[] segmentUpdateDetails = updateStatusManager.getUpdateStatusDetails();
```
Comment (Indhumathi27) [inline core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java:475]: Please add comments why this change is required

Comment (Zhangshunyu) [reply core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java:475]: @Indhumathi27 added

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java (hunk 2 above)
Comment (Zhangshunyu) [reply core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java]: @marchpure OK

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1605/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3345/

**Comment by Zhangshunyu** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1612/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3353/

**Comment by jackylk** [comment]
LGTM
